### PR TITLE
Fix deb package java dependencies

### PR DESCRIPTION
**Pull request by alrick87**

### CODE REVIEW

File: deb-packaging/src/deb/control/postrm [hunk 1]
```diff
@@ -15,4 +15,6 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-/usr/sbin/userdel brooklyn
+if /usr/bin/getent passwd brooklyn; then
+    /usr/sbin/userdel brooklyn
```
Comment (johnmccabe) [inline deb-packaging/src/deb/control/postrm]: Do we really want to purge the user? Looking at postgresql and mysql they don't appear to delete the users when you remove/purge.

Comment (alrick87) [inline deb-packaging/src/deb/control/postrm]: @johnmccabe If you think we shouldn't delete the user, I will delete the whole postrm file then.

Comment (m4rkmckenna) [inline deb-packaging/src/deb/control/postrm]: IMO its better to leave the user alone ... what if it had been created before install?

Comment (alrick87) [inline deb-packaging/src/deb/control/postrm]: OK

**Comment by johnmccabe** [comment]
@alrick87 I don't think its that simple, if you look at what mysql does for example they actually check if any owned files still exist on the system before purging.

**Comment by alrick87** [comment]
@johnmccabe dpkg checks if the directories with config files are empty and if not it doesn't delete them. I thought it is enough.

**Comment by johnmccabe** [comment]
what is the user being used for?

**Comment by alrick87** [comment]
@johnmccabe in this iteration for nothing, as I didn't find an easy way to run brooklyn as a non-root user within the init script.

**Comment by johnmccabe** [comment]
Tried building and noticed the following:
- deb package is not being output to `target` dir in the deb-packaging module, instead its output to brooklyn-dist/target
You should edit the pom.xml and change:
```
<deb>target/apache-brooklyn_${project.version}_all.deb</deb>
```
to
```
<deb>${basedir}/target/apache-brooklyn_${project.version}_all.deb</deb>
```

**Comment by johnmccabe** [comment]
Also, notice that debug logging appears to be on by default in the logback.xml, I think debug should be off by default, INFO level and above seems sensible. @aledsage wdyt

**Comment by m4rkmckenna** [comment]
@johnmccabe @alrick87 
```
<deb>${project.build.directory}/apache-brooklyn_${project.version}_all.deb</deb>
```

**Comment by johnmccabe** [comment]
thanks @m4rkmckenna I'd just picked the var in use elsewhere in the `pom.xml`, looks like there are other occurrences of `${basedir}` throughout brooklyn that'll need addressed separately to this PR.

**Comment by alrick87** [comment]
@m4rkmckenna @johnmccabe 
thanks, I'll make an update in a moment

**Comment by johnmccabe** [comment]
Also observe the following deploying on `15.10`, suspect the same on xenial.

Which distros did you test the change on?
```
sudo gdebi apache-brooklyn_0.10.0-SNAPSHOT_all.deb
Reading package lists... Done
Building dependency tree
Reading state information... Done
Reading state information... Done
Requires the installation of the following packages: ca-certificates-java default-jre-headless java-common libasyncns0 libflac8 libogg0 libpcsclite1 libpulse0 libsctp1 libsndfile1 libvorbis0a libvorbisenc2 lksctp-tools openjdk-7-jre-headless tzdata-java

Apache Brooklyn is a framework for modeling, monitoring, and managing applications through autonomic blueprints.
Do you want to install the software package? [y/N]:y
Get:1 http://archive.ubuntu.com/ubuntu/ wily/main libasyncns0 amd64 0.8-5build1 [12.3 kB]
Get:2 http://archive.ubuntu.com/ubuntu/ wily/main libogg0 amd64 1.3.2-1 [17.2 kB]
Get:3 http://archive.ubuntu.com/ubuntu/ wily/main libsctp1 amd64 1.0.16+dfsg-2 [8414 B]
Get:4 http://archive.ubuntu.com/ubuntu/ wily/main libvorbis0a amd64 1.3.4-2 [87.3 kB]
Get:5 http://archive.ubuntu.com/ubuntu/ wily/main libvorbisenc2 amd64 1.3.4-2 [70.4 kB]
Get:6 http://archive.ubuntu.com/ubuntu/ wily/main java-common all 0.52 [131 kB]
Get:7 http://archive.ubuntu.com/ubuntu/ wily/main default-jre-headless amd64 2:1.7-52 [4086 B]
Get:8 http://archive.ubuntu.com/ubuntu/ wily/main ca-certificates-java all 20140324 [12.3 kB]
Get:9 http://archive.ubuntu.com/ubuntu/ wily/main tzdata-java all 2015g-1 [69.1 kB]
Get:10 http://archive.ubuntu.com/ubuntu/ wily/main libpcsclite1 amd64 1.8.14-1ubuntu1 [21.4 kB]
Get:11 http://archive.ubuntu.com/ubuntu/ wily/main libflac8 amd64 1.3.1-4 [210 kB]
Get:12 http://archive.ubuntu.com/ubuntu/ wily-updates/main libsndfile1 amd64 1.0.25-9.1ubuntu0.15.10.1 [138 kB]
Get:13 http://archive.ubuntu.com/ubuntu/ wily/main libpulse0 amd64 1:6.0-0ubuntu13 [243 kB]
Err http://archive.ubuntu.com/ubuntu/ wily-updates/main openjdk-7-jre-headless amd64 7u95-2.6.4-0ubuntu0.15.10.1
  404  Not Found
Get:14 http://archive.ubuntu.com/ubuntu/ wily/main lksctp-tools amd64 1.0.16+dfsg-2 [41.9 kB]
Err http://security.ubuntu.com/ubuntu/ wily-security/main openjdk-7-jre-headless amd64 7u95-2.6.4-0ubuntu0.15.10.1
  404  Not Found
Fetched 1067 kB in 0s (0 B/s)
Error during install: 'Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/o/openjdk-7/openjdk-7-jre-headless_7u95-2.6.4-0ubuntu0.15.10.1_amd64.deb 404  Not Found
```

**Comment by alrick87** [comment]
@johnmccabe please run "sudo apt-get update first" then you won't experience any 404 errors (at least that worked for me on all the Ubuntu distros)

**Comment by alrick87** [comment]
> Also, notice that debug logging appears to be on by default in the logback.xml, I think debug should be off by default, INFO level and above seems sensible.

@johnmccabe I think changing logging level is out of scope for this PR

**Comment by johnmccabe** [comment]
Regarding running as non-root on Ubuntu Trusty 14.04 LTS, it ships with upstart, I tried locally with the following and it does the job, respawning on failure, running as brooklyn etc. Is there a reason why upstart hasn't been considered for 14.04?:
```bash
description "rough upstart script for apache-brooklyn"

start on started networking
stop on runlevel [016]
respawn
respawn limit 5 10

setuid brooklyn
setgid brooklyn

console output

pre-start script
    logger -i -t "$UPSTART_JOB" "[`date -u +%Y-%m-%dT%T.%3NZ`] Starting Apache Brooklyn"
end script

script
BROOKLYN_HOME=/opt/brooklyn/
JAVA_OPTS="-Dbrooklyn.location.localhost.address=127.0.0.1 -Xms256m -Xmx1g -XX:MaxPermSize=256m -XX:+UseConcMarkSweepGC"
INITIAL_CLASSPATH=${BROOKLYN_HOME}/conf:${BROOKLYN_HOME}/lib/patch/*:${BROOKLYN_HOME}/lib/wars:${BROOKLYN_HOME}/lib/brooklyn/*:${BROOKLYN_HOME}/lib/dropins/*

export BROOKLYN_HOME JAVA_OPTS INITIAL_CLASSPATH
exec java ${JAVA_OPTS} -cp "${INITIAL_CLASSPATH}" org.apache.brooklyn.cli.Main launch
end script

pre-stop script
  logger -i -t "$UPSTART_JOB" "[`date -u +%Y-%m-%dT%T.%3NZ`] Stopping Apache Brooklyn"
end script
```

**Comment by alrick87** [comment]
@johnmccabe I went with SysV to cover all the Ubuntu and Debian versions with one script. If we go with Upstart, we will also have to make another package with systemd script for Ubuntu starting from 15.04 and Debian 8

**Comment by johnmccabe** [comment]
@aledsage @alrick87 I think having 2 debian packages is acceptable, when we get repos up and running it'll be transparent to the end user. wdyt?

**Comment by alrick87** [comment]
@johnmccabe I'm in favor with any solution which the majority will accept. Also I think having two separate packages is easier to create and maintain. Plus I can reuse the upstart script to have an RPM package for Centos 6.

**Comment by johnmccabe** [comment]
@alrick87 re:Centos6 you'd need to check the version of upstart supplied, you may have have feature parity between the CentOS and Ubuntu releases

**Comment by neykov** [comment]
Changes look good. Merging.
Having a single `.deb` sounds good to me. Easier to support for us and less confusion for users.

**Pull request merged**